### PR TITLE
[utils] Optimize CFileItem::IsPicture and URIUtils::HasExtension.

### PR DESCRIPTION
**Pull request by ksooo**
Optimize `CFileItem::IsPicture` to avoid super expensive calls to `URIUtils::HasExtension`. Optimize `URIUtils::HasExtension` implementation,

Before:

Screenshot 2022-07-28 at 11 01 48

After:

Screenshot 2022-07-28 at 14 55 23

Runtime-tested on Android and macOS.
Profiled on macOS.

@phunkyfish when you find some time for a review.

### CODE REVIEW

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -1057,7 +1057,14 @@ bool CFileItem::IsPicture() const
   if (HasVideoInfoTag())
     return false;
 
-  return CUtil::IsPicture(m_strPath);
+  if (HasPVRTimerInfoTag() || HasPVRChannelInfoTag() || HasPVRChannelGroupMemberInfoTag() ||
+      HasPVRRecordingInfoTag() || HasEPGInfoTag() || HasEPGSearchFilter())
+    return false;
+
+  if (!m_strPath.empty())
+    return CUtil::IsPicture(m_strPath);
```
Comment (garbear) [inline xbmc/FileItem.cpp:1065]: Why not let CUtil::IsPicture() handle the empty case? Is it not expected to return false immediately if the path is empty?

Comment (ksooo) [reply xbmc/FileItem.cpp:1065]: I wanted to avoid the additional function call.

**Comment by fuzzard** [comment]
jenkins build this please

File: xbmc/FileItem.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/FileItem.cpp:1065]: Fair.

**Review by garbear — APPROVED** [review]
It looks like there's still a _lot_ of baby allocations occurring with all the `std::string` and `std::vector` but leaving nice c++ code for some low-level code is probably diminishing returns at this point.

**Comment by vpeter4** [comment]
If I'm not mistaken this PR breaks installing addons manually from ZIP file.

OK version:  KodiSetup-20220816-3f1399ef-master-x64.exe
NOK version: KodiSetup-20220817-8a85a4c4-master-x64.exe

**Pull request merged**